lib/grandpa/vote_message.go

```diff
@@ -126,51 +126,67 @@ func (s *Service) validateVoteMessage(from peer.ID, m *VoteMessage) (*Vote, erro
 	// check for message signature
 	pk, err := ed25519.NewPublicKey(m.Message.AuthorityID[:])
 	if err != nil {
+		// TODO: affect peer reputation
 		return nil, err
 	}
 
 	err = validateMessageSignature(pk, m)
 	if err != nil {
+		// TODO: affect peer reputation
 		return nil, err
 	}
 
 	if m.SetID != s.state.setID {
 		return nil, ErrSetIDMismatch
 	}
 
-	// check that vote is for current round
-	if m.Round != s.state.round {
-		if m.Round < s.state.round {
-			// peer doesn't know round was finalised, send out another commit message
-			header, err := s.blockState.GetFinalisedHeader(m.Round, m.SetID)
-			if err != nil {
-				return nil, err
-			}
+	const maxRoundsLag = 1
+	minRoundAccepted := s.state.round - maxRoundsLag
+	if minRoundAccepted > s.state.round {
+		// we overflowed below 0 so set the minimum to 0.
+		minRoundAccepted = 0
+	}
 
-			cm, err := s.newCommitMessage(header, m.Round)
-			if err != nil {
-				return nil, err
-			}
+	const maxRoundsAhead = 1
+	maxRoundAccepted := s.state.round + maxRoundsAhead
```

Comment:
Do we need to consider overflow here, or will round not get high enough for that?

Reply:
Good question. I doubt the round would ever reach the max uint64. It also gets reset on authority set changes. The spec defines it as an `uint64` as well. I'll link this up on Slack see if anyone thinks anything on this.

Reply:
From Elizabeth:

> nah I don't think so. that's a LOT of rounds